src/main/java/org/prebid/server/bidder/appush/AppushBidder.java

```diff
@@ -0,0 +1,158 @@
+            builder.type(PUBLISHER_PROPERTY).placementId(extImpAppush.getPlacementId());
+        } else if (StringUtils.isNotEmpty(extImpAppush.getEndpointId())) {
+            builder.type(NETWORK_PROPERTY).endpointId(extImpAppush.getEndpointId());
+        }
+
+        return builder.build();
+    }
+
+    private ExtImpAppush parseExtImp(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), APPUSH_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private HttpRequest<BidRequest> makeHttpRequest(BidRequest request, Imp imp) {
+        final BidRequest outgoingRequest = request.toBuilder().imp(List.of(imp)).build();
+
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .payload(outgoingRequest)
+                .body(mapper.encodeToBytes(outgoingRequest))
+                .build();
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            List<BidderBid> bids = extractBids(httpCall.getRequest().getPayload(), bidResponse);
+            Result.empty();
+            return Result.withValues(bids);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .filter(Objects::nonNull)
+                .map(bid -> BidderBid.of(bid, getBidType(bid.getImpid(), bidRequest.getImp()), bidResponse.getCur()))
+                .toList();
+    }
+
+    private static BidType getBidType(String impId, List<Imp> imps) {
+        for (Imp imp : imps) {
+            if (imp.getId().equals(impId)) {
+                if (imp.getBanner() != null) {
+                    return BidType.banner;
+                } else if (imp.getVideo() != null) {
+                    return BidType.video;
+                } else if (imp.getXNative() != null) {
+                    return BidType.xNative;
+                }
+                break;
+            }
+        }
+
+        throw new PreBidException(String.format(FAILED_TO_FIND_IMPRESSION_ERROR_MESSAGE, impId));
```

Comment:
WE don't need FAILED_TO_FIND_IMPRESSION_ERROR_MESSAGE constant, pls just inline this